src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            grow();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (checkArraySize(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " not found. "
+                    + "Actual array size: " + size);
+        }
+        if (size == list.length) {
+            grow();
+        }
+        System.arraycopy(list, index, list, index + 1, size - index);
+        list[index] = value;
+        size++;
+    }
 
+    private boolean checkArraySize(int index) {
+        return index < 0 || index > size;
     }
```

Comment:
```suggestion
    @Override
    public void add(T value, int index) {
       checkIndexForAdding(index);
        if (size == list.length) {
            grow();
        }
        System.arraycopy(list, index, list, index + 1, size - index);
        list[index] = value;
        size++;
    }

    private void checkIndexForAdding(int index) {
         if (index < 0 || index > size) {
         	throw new ArrayListIndexOutOfBoundsException("Index " + index + " not found. "
                    + "Actual array size: " + size);
        }
    }
```

і приватні методу мають бути під публічними